lib/modules/profiler/index.js

```diff
@@ -0,0 +1,56 @@
+const asciiTable = require('ascii-table');
+
+class Profiler {
+  constructor(embark) {
+    this.embark = embark;
+    this.logger = embark.logger;
+    this.events = embark.events;
+    this.plugins = embark.plugins;
+
+    this.registerConsoleCommand();
+  }
+
+  profile(contractName, contract) {
+    const self = this;
+    let table = new asciiTable(contractName);
+    table.setHeading('Function', 'Payable', 'Mutability', 'Inputs', 'Outputs');
+    contract.abiDefinition.forEach((abiMethod) => {
+      switch(abiMethod.type) {
+        case "constructor": 
+          table.addRow("constructor", abiMethod.payable, abiMethod.stateMutability, this.formatParams(abiMethod.inputs), this.formatParams(abiMethod.outputs));
+          break;
+        default:
+          table.addRow(abiMethod.name, abiMethod.payable, abiMethod.stateMutability, this.formatParams(abiMethod.inputs), this.formatParams(abiMethod.outputs));
+      }
+    });
+    self.logger.info(table.toString());
+  }
+
+  formatParams(params) {
+    if (undefined !== params && params.length) {
```

Comment:
You can simply do `if (params &params.length)` if you want

Reply:
Also, It's prettier to do 
```
if (!params || !params.length) {
  return "()";
}
// Do other stuff
```

Reply:
fixing.